Src/WitsmlExplorer.Api/Workers/AnalyzeGapWorker.cs

```diff
@@ -0,0 +1,181 @@
+    }
+
+    private (WorkerResult, RefreshAction) GetGapReportResult(AnalyzeGapJob job, List<AnalyzeGapReportItem> gapReportItems, bool isDepth, string logUid)
+    {
+        Logger.LogInformation("Analyzing gaps is done. {jobDescription}", job.Description());
+        job.JobInfo.Report = GetGapReport(gapReportItems, job.LogReference, isDepth);
+        WorkerResult workerResult = new(GetTargetWitsmlClientOrThrow().GetServerHostname(), true, $"Analyze gaps for log: {logUid}", jobId: job.JobInfo.Id);
+        return (workerResult, null);
+    }
+
+    private AnalyzeGapReport GetGapReport(List<AnalyzeGapReportItem> analyzeGapItems, LogObject logReference, bool isDepthLog)
+    {
+        return new AnalyzeGapReport
+        {
+            Title = $"Analyze gaps report",
+            Summary = analyzeGapItems.Count > 0
+                ? $"Found {analyzeGapItems.Count} gaps for {(isDepthLog ? "depth" : "time")} log '{logReference.Name}':"
+                : "No gaps were found for selected items.",
+            LogReference = logReference,
+            ReportItems = analyzeGapItems
+        };
+    }
+
+    /// <summary>
+    /// Get all gaps for input list of mnemonic values larger then requested gap size.
+    /// </summary>
+    /// <param name="mnemonic">Name of mnemonic.</param>
+    /// <param name="inputList">Input list of tuple mnemonic values.</param>
+    /// <param name="requestedGapSize">Requested gap size defined by user.</param>
+    /// <param name="isLogIncreasing">Is log direction increasing.</param>
+    /// <returns>Report items with gap size information.</returns>
+    private IEnumerable<AnalyzeGapReportItem> GetAnalyzeGapReportItem(string mnemonic, IList<Tuple<Index, string>> inputList, Index requestedGapSize, bool isLogIncreasing)
+    {
+        List<AnalyzeGapReportItem> gapValues = new();
+        Index startGapIndex = null;
+        bool insideGap = false;
+        
+        for (int i = 0; i < inputList.Count; i++)
+        {
+            var typeItem =  inputList[i].Item1;
+            string value = inputList[i].Item2;
+            if (string.IsNullOrEmpty(value))
+            {
+                if (!insideGap)
+                {
+                    startGapIndex = i > 0 ? (inputList[i - 1].Item1) : typeItem;
+                    insideGap = true;
+                }
+            }
+            else
+            {
+                if (insideGap)
+                {
+                    var gapSize = isLogIncreasing ? (typeItem - startGapIndex) : (startGapIndex - typeItem) ;
+                    if (gapSize >= requestedGapSize)
+                    {
+                        gapValues.Add(new AnalyzeGapReportItem
+                        {
+                            Mnemonic = mnemonic,
+                            Start = startGapIndex.ToString(),
+                            End = typeItem.ToString(),
+                            GapSize = gapSize.ToString(),
```

Comment:
Could the gapSize be rounded to 3 digits of precision for depth logs? Otherwise the result has too many unnecessary digits such as "1.2199999999999989".

Reply:
fixed